ed25519/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+extern crate ewasm_api;
+extern crate ed25519_dalek;
+extern crate sha2;
+
+use ed25519_dalek::{Signature,PublicKey};
+use sha2::Sha512;
+
+#[no_mangle]
+pub extern "C" fn main() {
+    let mut length = ewasm_api::calldata_size();
+    let mut zero_len: usize = 0;
+
+    if length < 128 {
+      zero_len = 128 - length;
+    } else {
+      length = 128;
+    }
+
+    // todo: consume flat gas cost
+    ewasm_api::consume_gas(2000);
+
+    let mut input = ewasm_api::unsafe_calldata_copy(0, length);
+    input.append(&mut vec![0; zero_len]);
+
+    let message = &input[..32];
+    let output = vec![0u8; 4];
+
+    let sig = match Signature::from_bytes(&input[32..64]) {
```

Comment:
Can you create another module which takes the input, zero pads it and returns the output buffer or an error?

Then have unit tests in that module and use it here.

See how the `ecadd` precompile is done.

Reply:
Yeah.

Reply:
By another module i just mean another file, it doesn't need to be another git repo :)